web/src/pages/Main.tsx

```diff
@@ -1,14 +1,65 @@
 import { Header } from '../components/common/Header'
-import { useContext } from 'react'
-import { AuthContext } from 'react-oauth2-code-pkce'
+import { useEffect, useState } from 'react'
+import { MoleTable } from '../components/feature/MoleTable'
+import MercuryAPI from '../api/MercuryAPI'
+import { AxiosResponse } from 'axios'
+
+// TODO: Get type from generated API
+export type TComponentResponse = {
+  [key: string]: { [key: string]: string }
+}
 
 export const MainPage = (): JSX.Element => {
-  const { token } = useContext(AuthContext)
+  const [components, setComponents] = useState<TComponentResponse>()
+  const [loading, setLoading] = useState<boolean>(false)
+
+  // Fetch list of components name once on page load
+  useEffect(() => {
+    const mercuryApi = new MercuryAPI('')
+    mercuryApi
+      .getComponents()
+      .then((response: AxiosResponse<TComponentResponse>) =>
+        setComponents(response.data)
+      )
+      .finally(() => setLoading(false))
+  }, [])
+
+  // TODO: Remove with actual calculation results
+  const multiflashResult = {
+    phase_values: {
+      Vapor: { percentage: 0.9989992588, mercury: 104.82781237 },
+      Mercury: { percentage: 0.0010007412, mercury: 1000000000.0 },
+    },
+    component_fractions: {
+      2: [0.01693, 0.000000000000000000000003431],
+      1: [0.05399, 0.000000000000000000000006807],
+      101: [0.8563, 0.0000000000000000000001379],
+      201: [0.04577, 0.00000000000000000000000481],
+      301: [0.01763, 0.000000000000000000000001301],
+      401: [0.002424, 0.0000000000000000000000001389],
+      402: [0.004378, 0.0000000000000000000000002324],
+      503: [0.001212, 0.00000000000000000000000005006],
+      504: [0.001102, 0.00000000000000000000000004238],
+      608: [0.0001473, 0.000000000000000000000000005109],
+      710: [0.00008415, 0.000000000000000000000000002175],
+      809: [0.00002304, 0.000000000000000000000000000433],
+      5: [0.00000001236, 1.0],
+    },
+  }
+
+  if (loading) return <>Loading...</>
```

Comment:
I prefer this `{isLoading && <>Loading...</>}` and trying to avoid having multiple returns in a component because it's easier to control what the component should return.

Reply:
Disagree. Having multiple nested conditional inline if-else renders in a single return statement will be less readable/maintanable.

Reply:
Okay, we'll go with that, then. As long as the file isn't too big and the returns are easy to find, I guess it won't be a problem.